lucene/core/src/java/org/apache/lucene/search/LRUQueryCache.java

```diff
@@ -448,13 +459,42 @@ void assertConsistent() {
     }
   }
 
+  // pkg-private for testing
+  // return the list of queries being loaded asynchronously
+  List<Query> inFlightQueries() {
+    lock.lock();
+    try {
+      return new ArrayList<>(inFlightAsyncLoadQueries);
```

Comment:
I'm still confused about `lock` -- do we always hold the lock when checking if query is already in the map?  If so, we don't need a `ConcurrentHashMap`?  If not, why do we even have the `lock` since it is a `ConcurrentHashMap`.

Reply:
Not all places which access `inFlightAsyncLoadQueries` take a lock -- the main being `cacheAsynchronously`

Reply:
This specific access site does not lead a lock -- thanks for highlighting that, fixed!